### PR TITLE
PULSEBS-17 backend implementation

### PR DESCRIPTION
**Pull request by Francexi**

### CODE REVIEW

File: server/API/REST.php [hunk 1]
```diff
@@ -634,6 +634,85 @@ function cancel_booking($vars) {
...
+				}
+				else{
+					$sql .= $year_subsql.$semester_subsql;
+				}
+
+				$sql .= ")";
+			}
+
+			if(null !== $end_time){
+				$sql .= " AND end_ts <= ".$end_time->getTimestamp();
```
Comment (LeaX-XIV) [inline server/API/REST.php]: lecture start------------end time------------lecture end
In this case, we want the lecture to be considered for the update, so consider start_ts, not end_ts.

```suggestion
				$sql .= " AND start_ts <= ".$end_time->getTimestamp();
```

File: server/endpoints.md [hunk 2]
```diff
@@ -177,6 +177,21 @@ Requires login as teacher
   - _response body_
     - success: bool
 
+# Edit a block of lecture online status
+
+Requires login as support officer
+
+- **PATCH** /api/lectures/online
+  - _request params_
+    - value: bool
```
Comment (LeaX-XIV) [inline server/endpoints.md]: Define better name for value, or dedscribe behavior if true/false

**Review by LeaX-XIV — CHANGES_REQUESTED** [review]

File: server/endpoints.md (hunk 2 above)
Comment (Francexi) [reply server/endpoints.md]:
```suggestion
    - value: bool (true online, false offline)
```

Comment (LeaX-XIV) [reply server/endpoints.md]:
```suggestion
    - value: bool (true -> turn all to online, false -> turn all to presence)
```

**Review by LeaX-XIV — APPROVED** [review]

**Pull request merged**